mlflow/store/artifact/optimized_s3_artifact_repo.py

```diff
@@ -50,7 +50,7 @@ def __init__(
         secret_access_key=None,
         session_token=None,
         credential_refresh_def=None,
-        addressing_style="path",
+        addressing_style="virtual",
```

Comment:
I think we should actually default to the boto3 default value in https://boto3.amazonaws.com/v1/documentation/api/1.27.0/guide/configuration.html#:~:text=addressing_style%3A%20The%20S3%20addressing%20style.%20When%20necessary%2C%20Boto%20automatically%20switches%20the%20addressing%20style%20to%20an%20appropriate%20value.%20The%20following%20values%20are%20supported.

So either pass `None` or `"auto"` here depending on what the default is in the actual Python API doc